Exiled.API/Features/Player.cs

```diff
@@ -1930,18 +1982,39 @@ public void Kill(string deathReason, string cassieAnnouncement = "")
             if ((Role.Side != Side.Scp) && !string.IsNullOrEmpty(cassieAnnouncement))
                 Cassie.Message(cassieAnnouncement);
 
-            ReferenceHub.playerStats.KillPlayer(new CustomReasonDamageHandler(deathReason, float.MaxValue, cassieAnnouncement));
+            ReferenceHub.playerStats.KillPlayer(new CustomReasonDamageHandler(deathReason, -1, cassieAnnouncement));
+        }
+
+        /// <summary>
+        /// Kills the player and vaporizes the body.
+        /// </summary>
+        /// <param name="attacker">The <see cref="Player"/> attacking player.</param>
+        /// <param name="cassieAnnouncement">The cassie announcement to make upon death.</param>
+        public void Vaporize(Player attacker = null, string cassieAnnouncement = "")
+        {
+            if ((Role.Side != Side.Scp) && !string.IsNullOrEmpty(cassieAnnouncement))
+                Cassie.Message(cassieAnnouncement);
+
+            Kill(new DisruptorDamageHandler(attacker?.Footprint ?? Footprint, -1));
```

Comment:
This will make them kill themselves, rather than it being logged as the server killing them